### ISSUE TITLE
Compile static libraries with -fPIC

### ISSUE THREAD

**Issue by wesm** [conda-forge/boost-feedstock#11]
Similar to other C++ libraries where we've addressed static linking issues, boost also needs `-fPIC` in its CXXFLAGS to be able to be statically linked when creating other shared libraries. It turns out that Ubuntu 14.04's boost also does not do this, and I just got this error in another build:

```shell
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libboost_filesystem.a(path.o): relocation R_X86_64_32S against `_ZNSs4_Rep20_S_empty_rep_storageE' can not be used when making a shared object; recompile with -fPIC
/usr/lib/x86_64-linux-gnu/libboost_filesystem.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
```

To be clear, the use case is creating `libfoo.so` with its dependencies statically linked (so users of `libfoo` do not need to install its transitive dependencies as other shared libraries). Is there some other way around this other than always compiling with `-fPIC` (the known solution)?

**Comment by ocefpaf** [comment]
@wesm I am having this discussion (use `-fPIC`) with a colleague on other libraries and we decided that we should always enable it. We could not find a downside and it solves tons of problems. Also, as you mentioned for Ubuntu, `-fPIC` is enabled by default by most Linuxes distros.

:+1:

**Comment by wesm** [comment]
addressed in #12
